test/CoinSelection/RoundRobin.purs

```diff
@@ -0,0 +1,211 @@
+module Test.Ctl.CoinSelection.RoundRobin where
+
+import Prelude
+
+import Ctl.Internal.BalanceTx.CoinSelection (SelectionState, runRoundRobinM)
+import Ctl.Internal.Test.TestPlanM (TestPlanM)
+import Ctl.Internal.Types.TokenName (TokenName)
+import Data.Foldable (all, foldl, sum)
+import Data.Function (on)
+import Data.Generic.Rep (class Generic)
+import Data.Identity (Identity(Identity))
+import Data.List
+  ( List
+  , fromFoldable
+  , length
+  , singleton
+  , sort
+  , sortBy
+  , tail
+  , zip
+  , (:)
+  )
+import Data.Map as Map
+import Data.Maybe (Maybe(Just, Nothing), fromMaybe, maybe)
+import Data.Newtype (class Newtype, unwrap)
+import Data.NonEmpty (NonEmpty)
+import Data.Ordering (invert)
+import Data.Set as Set
+import Data.Show.Generic (genericShow)
+import Data.Traversable (sequence)
+import Data.Tuple (snd, swap, uncurry)
+import Data.Tuple.Nested (type (/\), (/\))
+import Data.Unfoldable (replicateA)
+import Effect.Aff (Aff)
+import Mote (group, test)
+import Test.QuickCheck (Result, (<?>), (===))
+import Test.QuickCheck.Arbitrary (class Arbitrary, arbitrary)
+import Test.QuickCheck.Gen (Gen, chooseInt)
+import Test.Spec.Assertions (shouldEqual)
+import Test.Spec.QuickCheck (quickCheck)
+
+suite :: TestPlanM (Aff Unit) Unit
+suite = do
+  group "Grouping and ungrouping" do
+    test "groupByKey_ungroupByKey" $
+      quickCheck
+        \(x :: List (Int /\ Int)) -> on (===) sort
+          x
+          (ungroupByKey $ groupByKey x)
+    test "ungroupByKey_groupByKey" $
+      quickCheck
+        -- There is no Arbiratry for Map
+        \(x :: List (Int /\ (NonEmpty List Int))) ->
+          let
+            m = (map fromFoldable $ Map.fromFoldable x)
+          in
+            on (===) (map sort)
+              m
+              (groupByKey $ ungroupByKey m)
+  group "RoundRobin" do
+    -- TODO: doc
+    test "identity"
```

Comment:
I would extract `identity` and `iterationCount` properties to separate functions, as this would make the code cleaner